src/NuGet.Core/NuGet.Configuration/Settings/Settings.cs

```diff
@@ -575,10 +590,20 @@ private static ISettings LoadSettingsGivenSettingsFiles(List<SettingsFile> setti
                     throw new InvalidOperationException(message);
                 }
 
-                userSpecificSettings = ReadSettings(rootDirectory, configFileName, settingsLoadingContext: settingsLoadingContext);
+                yield return ReadSettings(rootDirectory, configFileName, settingsLoadingContext: settingsLoadingContext);
             }
+        }
+
+        private static string GetUserSettingsDirectory(string rootDirectory, bool useTestingGlobalPath)
+        {
+            return useTestingGlobalPath
+                ? Path.Combine(rootDirectory, "TestingGlobalPath")
+                : NuGetEnvironment.GetFolderPath(NuGetFolderPath.UserSettingsDirectory);
+        }
 
-            return userSpecificSettings;
+        private static string GetAdditionalUserWideConfigurationDirectory(string userSettingsDirectory)
+        {
+            return Path.Combine(userSettingsDirectory, "config");
```

Comment:
Can you move the config folder name to const or read-only property? The suggestion below could not be correct.

```suggestion
            return Path.Combine(userSettingsDirectory, ConfigurationConstants.AdditionalUserConfigsFolderName);
```